### PR TITLE
Add FXIOS-2253 [vXXX] IPv6 addresses not identified as valid URLs

### LINKED ISSUE #6015: IPv6 addresses not identified as valid URLs
Steps to reproduce:

Navigate to an IPv6 address like "http://[2a01:4f8:c0c:4925::10]".

Actual results:

Firefox submits the address to Google as a search term.

Expected results:

The browser should have navigated to the IPv6 address.

Via  : https://bugzilla.mozilla.org/show_bug.cgi?id=1608732

Notes:
- This is not reproducible on Chrome. 
- Reproducible in Safari.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-2253)

### PR DESCRIPTION
**Pull request by q2r5**
Fixes #6015.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1065)

### CODE REVIEW

**Comment by lmarceau** [comment]
@q2r5 Would you mind rebasing this PR so I can trigger the Bitrise? Also, I tried navigating with the example `http://[2a01:4f8:c0c:4925::10]` from the ticket and wasn't successful. Do you remember any ipv6 address that was working on your side? Thanks a lot!

**Comment by lmarceau** [comment]
Ah sorry I pressed enter on my comment after you already had rebased haha. But my question regarding the example still applies 😄 Thanks!

**Comment by q2r5** [comment]
It seems the IP given in the original ticket doesn't resolve anymore. [https://[2620:fe::fe]/](https://[2620:fe::fe]/) should work though.

**Comment by lmarceau** [comment]
Ok perfect! I was using `http` instead of `https` and it wasn't working for me. I tried with `https://[2001:4860:4860::8888]/` and it works to navigate to `https://dns.google/`. Would an IPV6 with `http` work thought?

Also, there's one test failing: `testURIFixup()`
Screen Shot 2022-01-14 at 8 45 27 PM

**Comment by q2r5** [comment]
@lmarceau http://[2606:2800:220:1:248:1893:25c8:1946]/ should a valid IPv6 HTTP URL.

**Comment by lmarceau** [comment]
@q2r5 We should add some tests in our SearchTests.swift for the new cases we're covering. I think we can start with the same one implemented on this PR: https://github.com/brave/brave-ios/commit/1745bb07382de329c521d620b362858c17638704

File: ClientTests/SearchTests.swift [hunk 1]
```diff
@@ -46,7 +49,7 @@ class SearchTests: XCTestCase {
         checkInvalidURL("javascript:")
         checkInvalidURL("javascript:alert(%22hi%22)")
         checkInvalidURL("ftp:")
-        
+        checkInvalidURL("123:7::")
```
Comment (lmarceau) [inline ClientTests/SearchTests.swift:52]: Could we also add the following tests? They shouldn't be valid URL.
```
        checkInvalidURL("foo:5000")
        checkInvalidURL("http://::192.9.5.5")
        checkInvalidURL("http://::192.9.5.5:8080")
```

**Comment by lmarceau** [comment]
Hello! Just an update on this ticket. It will take a bit longer since we wanna have a deeper look at the implication of the changes for the URI fix up 👍

**Pull request closed without merging**